cpp/src/arrow/util/compression_test.cc

```diff
@@ -341,19 +335,63 @@ class CodecTest : public ::testing::TestWithParam<Compression::type> {
 };
 
 TEST_P(CodecTest, CodecRoundtrip) {
-  if (GetCompression() == Compression::BZ2) {
+  const auto compression = GetCompression();
+  if (compression == Compression::BZ2) {
     // SKIP: BZ2 doesn't support one-shot compression
     return;
   }
 
   int sizes[] = {0, 10000, 100000};
+
+  // create multiple compressors to try to break them
+  std::unique_ptr<Codec> c1, c2;
+  ASSERT_OK(Codec::Create(compression, &c1));
+  ASSERT_OK(Codec::Create(compression, &c2));
+
   for (int data_size : sizes) {
     std::vector<uint8_t> data = MakeRandomData(data_size);
-    CheckCodecRoundtrip(GetCompression(), data);
+    CheckCodecRoundtrip(c1, c2, data);
 
     data = MakeCompressibleData(data_size);
-    CheckCodecRoundtrip(GetCompression(), data);
+    CheckCodecRoundtrip(c1, c2, data);
+  }
+}
+
+TEST_P(CodecTest, SpecifyCompressionLevel) {
+  const auto compression = GetCompression();
+  // The compression level is codec specific.
+  int32_t compression_level;
+  switch (compression) {
+    case Compression::LZ4:
+    case Compression::LZO:
+    case Compression::UNCOMPRESSED:
+    case Compression::SNAPPY:
+      // Compression level cannot be specified for these
+      // compression types.
```

Comment:
Does it make sense to return an error if you specify a compression level for a compressor that does not support it? Either way we need tests asserting what the behavior is

Reply:
I decided to mention in the api for parquet-cpp that parquet/arrow do not do validation on the passed compression level.
1. it could be some effort to track what is the range of possible compression levels for a version of a compressor. Thus, it's best if the user finds out what's meaningful. Worst case initializing the codec would fail.
2. With that I also decided that I would not add validation for whether a compressor supports specifying a compression level. It's a hint and it doesn't necessarily lead to better compression ratio or speed.

I hope you're fine with that.